rts/Sim/Units/UnitHandler.cpp

```diff
@@ -321,6 +321,11 @@ void CUnitHandler::UpdateUnitMoveTypes()
 {
 	SCOPED_TIMER("Sim::Unit::MoveType");
 
+	for (activeUpdateUnit = 0; activeUpdateUnit < activeUnits.size(); ++activeUpdateUnit) {
```

Comment:
I don't think it's good to do this in a separate loop, you'll get a zillion cache misses.

Reply:
That is an issue. Units can collide with each other throughout the main loop so I'm not keen to set prevPos for each one individually. I can try it out though.